src/frontends/onnx/tests/onnx_import_convpool.in.cpp

```diff
@@ -507,6 +507,37 @@ OPENVINO_TEST(${BACKEND_NAME}, onnx_model_convtranspose_output_shape_auto_pads_s
     test_case.run();
 }
 
+// Test ConvTranspose with auto_pad='SAME_UPPER' WITHOUT explicit output_shape
+// This tests ONNX spec compliance: output = input * stride
+OPENVINO_TEST(${BACKEND_NAME}, onnx_model_convtranspose_auto_pad_same_upper_no_output_shape) {
+    // Input: [1,1,32,32], kernel=3x3, stride=1
+    // Expected output: [1,1,32,32] (per ONNX spec: output = input * stride = 32 * 1)
+    // Without the fix, OpenVINO produces [1,1,34,34] and this test fails
+    auto model = convert_model("convtranspose_auto_pad_same_upper_no_output_shape.onnx");
+
+    // Verify the output shape is correct
+    ASSERT_EQ(model->get_output_shape(0), (ov::Shape{1, 1, 32, 32}));
```

Comment:
You verify only the output shape in the tests, however we also need to make sure inference will be done correctly, please add tests for that

Reply:
Thank you for your comment. I added